src/main.ts

```diff
@@ -44,4 +46,4 @@ if (direction === 'push') {
     throw new Error(`Unknown direction ${direction}`);
 }
 
-core.setOutput('imageUrl', imageUrl);
+writeFileSync(process.env['GITHUB_STATE'], `imageUrl=${imageUrl}${EOL}`, { flag: 'a' });
```

Comment:
This is exactly [what core.setOutput() does](https://github.com/actions/toolkit/blob/ae38557bb0dba824cdda26ce787bd6b66cf07a83/packages/core/src/core.ts#L193-L196).  The `echo "::set-output name={name}::{value}"` style of set-output is being deprecated, but AFAIK `core.setOutput` should still be safe to call.

Reply:
[The announcement](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/) says that set-output and save-state are being deprecated. It does not say, however, that it does the same thing as setting the environment variable.

Replacing the call removes warnings in my pipeline. I see my error though where it should have been setting `GITHUB_OUTPUT` instead.

Reply:
Let me try pushing a release with an updated version of @actions/core, and see if that fixes your problem.

Reply:
Give v2 a try and see if that fixes it for you (because @actions/core DOES do exactly what this change does, so it shouldn't print a deprecation warning).  I've got an ECR image I need to push in the next couple of days, so I'll give it a go on my end too.

Reply:
Got an error.

```
node:internal/errors:497
    ErrorCaptureStackTrace(err);
    ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/home/runner/work/_actions/jwalton/gh-ecr-push/v2/dist/sourcemap-register.cjs' imported from /home/runner/work/_actions/jwalton/gh-ecr-push/v2/dist/index.js
    at new NodeError (node:internal/errors:406:5)
    at finalizeResolution (node:internal/modules/esm/resolve:233:11)
    at moduleResolve (node:internal/modules/esm/resolve:845:10)
    at defaultResolve (node:internal/modules/esm/resolve:1043:11)
    at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:383:12)
    at ModuleLoader.resolve (node:internal/modules/esm/loader:352:25)
    at ModuleLoader.getModuleJob (node:internal/modules/esm/loader:228:38)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:85:3[9](https://github.com/131db/cayman-medical-web/actions/runs/8928354081/job/24523942603#step:7:10))
    at link (node:internal/modules/esm/module_job:84:36) {
  url: 'file:///home/runner/work/_actions/jwalton/gh-ecr-push/v2/dist/sourcemap-register.cjs',
  code: 'ERR_MODULE_NOT_FOUND'
}

Node.js v20.8.1
```

Let me know if you need anything else.

Reply:
I forgot to rerun `npm run all`.  -_-  Should be fixed now.

Reply:
Works great! Thanks for the update!